src/Database/Expression/WindowExpression.php

```diff
@@ -119,6 +129,18 @@ public function order($fields)
             return $this;
         }
 
+        if ($fields instanceof Closure) {
+            $fields = $fields();
+            if (!($fields instanceof OrderByExpression)) {
+                throw new RuntimeException(
+                    'You must return an `OrderByExpression` from a Closure passed to `order()`.'
+                );
+            }
+            $this->order = $fields;
```

Comment:
This should act the same as when we add closure support to Query::order(). If we want it to only return fields there this needs to change.